File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/1567-maximum-number-of-vowels-in-a-substring-of-given-length.cpp
class Solution {
public:
    int maxVowels(string s, int k) {
        int n = s.size();
        int max {0};
        int count {0};
        for(int i = 0;i<k;i++){
            if(checkVowel(s[i])){
                count++;
            }
        }
        max = count>max ? count:max;
        for(int i = 1;i<=n-k;i++){
            if(checkVowel(s[i+k-1])){
                count++;
            }
            if(checkVowel(s[i-1])){
                count--;
            }
            max = count>max ? count:max;
        }
        return max;
    }

    bool checkVowel(char curr){
        if(curr == 'a' || curr == 'e' || curr == 'i' || curr == 'o' || curr == 'u'){
            return true;
        }
        return false;
    }
};

File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/1567-maximum-number-of-vowels-in-a-substring-of-given-length.java
class Solution {

    public int maxVowels(String s, int k) {
        int max = 0;
        int n = s.length();
        Queue<Character> stream = new LinkedList<>();
        int vC =0;
        for(int i = 0;i<n-k+1;i++){
            if(i==0){
                for(int j = 0;j<k;j++){
                    char curr = s.charAt(j);
                    if(curr=='a'||curr=='e'||curr=='i'||curr=='o'||curr=='u') vC++;
                    stream.add(curr);
                }
            }else{
                char out = stream.poll();
                if(out=='a'||out=='e'||out=='i'||out=='o'||out=='u') vC--;
                char p = s.charAt(i+k-1);
                if(p=='a'||p=='e'||p=='i'||p=='o'||p=='u') vC++;
                stream.add(p);
            }
            max = Math.max(vC,max);
        }
        return max;
    }
}